### PR TITLE
Use signed requests for attestation demo

### PR DESCRIPTION
**Pull request by arp102**
* Update attestation demo client to use signed requests for key wrapping/unwrapping
* Update pelz to accept signed requests
* Update demo script to generate keys and register a CA cert with pelz
* Improve demo error handling
* Replace placeholder demo input data with a new text file

### CODE REVIEW

**Review by bit-bender — COMMENTED** [review]
I left a couple comments for potentially incomplete memory clean-up that probably should really be addressed in kmyth.

In testing the demo, I am able to run it, but, for all of the JSON requests, I am getting the message `ERROR - Request signature did not match`.

I am still trying to figure out what I might be doing incorrectly, but it seems that this message results from a failed call to the kmyth function `ec_verify_buffer()` within the `validate_signature()` function in `src/util/enclave_request_signing.c`. Regardless of whether the invalid signature message is a result of my configuring and running the demo incorrectly or something else, I'm not sure why this "error" condition sets a "successful return" after printing this error.

File: attestation_demo/AppInitiator/App.cpp [hunk 1]
```diff
@@ -58,102 +60,269 @@
...
+    char *encoded_cert;
+    size_t encoded_cert_len;
     char *encoded_dek;
     size_t encoded_dek_len;
 
-    // TODO: 1. Change the message to a signed pelz request.
-    // TODO: 2. Change the message to a signed pelz request with individually encrypted fields.
-    // TODO: 3. Generate the request signature using a double-wrapped signing key (using kmyth).
+    if (read_bytes_from_file(key_path, &sign_key, &sign_key_len)) {
+        printf("failed to read the file at %s.\n", key_path);
```
[the remark this answers is not in the document]
Comment (bit-bender) [reply attestation_demo/AppInitiator/App.cpp:262]: I created [kmyth issue #184](https://github.com/NationalSecurityAgency/kmyth/issues/184). I think this is better resolved there.

File: attestation_demo/AppInitiator/App.cpp [hunk 2]
```diff
@@ -58,102 +60,269 @@
...
+        return -1;
+    }
+
+    if (read_bytes_from_file(cert_path, &sign_cert, &sign_cert_len)) {
+        printf("failed to read the file at %s.\n", cert_path);
+        free(sign_key);
+        return -1;
```
[the remark this answers is not in the document]
Comment (bit-bender) [reply attestation_demo/AppInitiator/App.cpp:269]: I created [kmyth issue #184](https://github.com/NationalSecurityAgency/kmyth/issues/184). I think this is better resolved there.

**Comment by arp102** [comment]
> In testing the demo, I am able to run it, but, for all of the JSON requests, I am getting the message `ERROR - Request signature did not match`.
> 
> I am still trying to figure out what I might be doing incorrectly, but it seems that this message results from a failed call to the kmyth function `ec_verify_buffer()` within the `validate_signature()` function in `src/util/enclave_request_signing.c`. Regardless of whether the invalid signature message is a result of my configuring and running the demo incorrectly or something else, I'm not sure why this "error" condition sets a "successful return" after printing this error.

It looks like the `ERROR - Request signature did not match` log message is incorrect, but we currently print it when signature validation is successful.
I can change the log message.

**Review by bit-bender — APPROVED** [review]

**Pull request merged**